### PR TITLE
Spring OAS 3.x support (complements https://github.com/swagger-api/swagger-codegen-generators/pull/618, closes https://github.com/swagger-api/swagger-codegen-generators/issues/444)

### PR DESCRIPTION
**Pull request by reta**
Springfox 3.0.0 (https://github.com/springfox/springfox/releases/tag/3.0.0) was finally released with OAS 3.0.x support so it unblocks the stubs generation for Spring-based servers and clients. 

Along with the OAS 3.x, an opportunity to modernize the Spring-based  generators has been taken (at some extent it was necessary because of  Springfox 3.0.0 baseline):
 - Update to Spring Framework 5.0.x line (4.3.x is about to be EOLed in October, https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Versions)
 - Update to Spring Boot 2.1.x line (1.5.x was EOLed last year, https://spring.io/blog/2018/07/30/spring-boot-1-x-eol-aug-1st-2019)
 - Update to Jackson 2.9.x line
 - Update to Jetty 9.3.x line

What was tested (for `spring`` library):
[x] spring-boot
[x] spring-cloud
[x] spring-mvc

Closes https://github.com/swagger-api/swagger-codegen-generators/issues/444, complements https://github.com/swagger-api/swagger-codegen-generators/pull/618

@HugoMario @rrockx-trifork @frantuma would appreciate guys if you could take a look, thanks!

### CODE REVIEW

File: src/main/java/io/swagger/codegen/v3/generators/java/SpringCodegen.java [hunk 1]
```diff
@@ -77,7 +77,7 @@
     protected boolean useOptional = false;
     protected boolean openFeign = false;
     protected boolean defaultInterfaces = true;
-    protected String springBootVersion = "1.5.22.RELEASE";
+    protected String springBootVersion = "2.1.16.RELEASE";
```
Comment (reta) [inline src/main/java/io/swagger/codegen/v3/generators/java/SpringCodegen.java:80]: 2.1.x is still supported for a few months at least (https://spring.io/blog/2019/12/10/spring-boot-2-1-x-eol-november-1st-2020)

File: src/main/resources/handlebars/JavaSpring/libraries/spring-boot/homeController.mustache [hunk 2]
```diff
@@ -10,7 +10,13 @@ import org.springframework.web.bind.annotation.RequestMapping;
 public class HomeController {
     @RequestMapping(value = "/")
     public String index() {
+        {{#useOas2}}
         System.out.println("swagger-ui.html");
         return "redirect:swagger-ui.html";
+        {{/useOas2}}
+        {{^useOas2}}
+        System.out.println("/swagger-ui/index.html");
```
Comment (reta) [inline src/main/resources/handlebars/JavaSpring/libraries/spring-boot/homeController.mustache:18]: The Swagger UI packaging is different for Springfox 2.x / 3.x release lines, accomodating this change

File: src/main/resources/handlebars/JavaSpring/libraries/spring-boot/pom.mustache [hunk 3]
```diff
@@ -100,6 +114,14 @@
             </dependency>
         {{/notNullJacksonAnnotation}}
 
+{{^useOas2}}
```
Comment (reta) [inline src/main/resources/handlebars/JavaSpring/libraries/spring-boot/pom.mustache:117]: I needed by Springfox 3.0.0

**Comment by reta** [comment]
Sorry @HugoMario , would you find some time to take a look? Thanks!

**Comment by HugoMario** [comment]
hey @reta, on it

File: src/main/resources/handlebars/JavaSpring/libraries/spring-boot/homeController.mustache (hunk 2 above)
Comment (HugoMario) [reply src/main/resources/handlebars/JavaSpring/libraries/spring-boot/homeController.mustache:18]: is the print statement required?

Comment (reta) [reply src/main/resources/handlebars/JavaSpring/libraries/spring-boot/homeController.mustache:18]: Not at all, it was there so I just preserved it for OAS 3.x as well, could be gone completely :)

**Comment by HugoMario** [comment]
thanks a lot @reta for this PR

**Comment by reta** [comment]
Thanks a lot for looking at it, @HugoMario!

**Pull request merged**